persistence/trees/module.go

```diff
@@ -24,35 +31,52 @@ func (*treeStore) Create(bus modules.Bus, options ...modules.TreeStoreOption) (m
 	return m, nil
 }
 
-func Create(bus modules.Bus, options ...modules.TreeStoreOption) (modules.TreeStoreModule, error) {
-	return new(treeStore).Create(bus, options...)
+func Create(bus modules.Bus, options ...modules.ModuleOption) (modules.Module, error) {
+	return new(TreeStore).Create(bus, options...)
 }
 
 // WithLogger assigns a logger for the tree store
-func WithLogger(logger *modules.Logger) modules.TreeStoreOption {
-	return func(m modules.TreeStoreModule) {
-		if mod, ok := m.(*treeStore); ok {
+func WithLogger(logger *modules.Logger) modules.ModuleOption {
+	return func(m modules.InjectableModule) {
+		if mod, ok := m.(*TreeStore); ok {
 			mod.logger = logger
 		}
 	}
 }
 
 // WithTreeStoreDirectory assigns the path where the tree store
 // saves its data.
-func WithTreeStoreDirectory(path string) modules.TreeStoreOption {
-	return func(m modules.TreeStoreModule) {
-		if mod, ok := m.(*treeStore); ok {
-			mod.treeStoreDir = path
+func WithTreeStoreDirectory(path string) modules.ModuleOption {
+	return func(m modules.InjectableModule) {
+		mod, ok := m.(*TreeStore)
+		if ok {
+			mod.TreeStoreDir = path
+		}
+	}
+}
+
+// WithTxIndexer assigns a TxIndexer for use during operation.
+func WithTxIndexer(txi indexer.TxIndexer) modules.ModuleOption {
+	return func(m modules.InjectableModule) {
+		mod, ok := m.(*TreeStore)
+		if ok {
+			mod.TXI = txi
 		}
 	}
 }
 
-func (t *treeStore) setupTrees() error {
-	if t.treeStoreDir == ":memory:" {
+func (t *TreeStore) GetModuleName() string  { return modules.TreeStoreModuleName }
+func (t *TreeStore) Start() error           { return nil }
```

Comment:
These are noop, `TreeStore` really seems like it should not be a `Module` but a `Submodule` like it was previously